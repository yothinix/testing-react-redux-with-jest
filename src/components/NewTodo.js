import React, { Component } from 'react'

class NewTodo extends Component {
  render() {
    return (
      <div className="new-todo">
        <input />
        <button className="btn btn-primary">Add</button>
      </div>
    )
  }
}

export default NewTodo
