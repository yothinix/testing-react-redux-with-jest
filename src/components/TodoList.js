import React from 'react'

import TodoItem from './TodoItem'

const TodoList = props => (
  <ul className="todo-list">
    (props.todos.map(todo =>
      <TodoItem key={todo.id} {...todo} />
    )}
  </ul>
)

export default TodoList
