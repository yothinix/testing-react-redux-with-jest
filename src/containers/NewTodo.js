import { connect } from 'react-redux'
import NewTodo from '../components/NewTodo'
import { addTodo } from '../actions/todos'

export default connect(null, { addTodo })(NewTodo)
