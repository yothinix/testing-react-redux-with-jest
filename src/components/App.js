import React from 'react'
import NewTodo from '../containers/NewTodo'
import TodoList from '../containers/TodoList'

const App = () => (
  <div className="todo-app">
    <NewTodo />
    <TodoList />
  </div>
)
export default App
