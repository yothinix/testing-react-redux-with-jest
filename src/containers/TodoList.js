import { connect } from 'react-redux'
import TodoList from '../components/TodoList'
import {
  selectTodos,
} from '../selectors/todosSelector'

export default connect(selectTodos)(TodoList)

