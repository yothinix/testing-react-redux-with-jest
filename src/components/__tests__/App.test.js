import React from 'react'
import { shallow } from 'enzyme'
import App from '../App'
import NewTodo from '../../containers/NewTodo'
import TodoList from '../../containers/TodoList'

describe('App', () => {
  let component

  beforeEach(() => {
    component = shallow(<App />)
  })

  it('mathces its snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('uses correct containers', () => {
    expect(component.find(NewTodo).exists()).toBe(true)
    expect(component.find(TodoList).exists()).toBe(true)
  })
})
