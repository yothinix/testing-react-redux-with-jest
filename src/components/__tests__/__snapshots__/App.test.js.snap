// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App mathces its snapshot 1`] = `
<div
  className="todo-app"
>
  <Connect(NewTodo) />
  <Connect(TodoList) />
</div>
`;
